go/app.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"html/template"
+	"net/http"
+	"os"
+)
+
+type User struct {
+	Id    string
+	Name  string
+	Email string
+}
+
+type HomePageData struct {
+	IntercomAppId    string
+	LoggedInUser     User
+	IntercomUserHash string
+}
+
+func ComputeHmac256(message string, secret string) string {
+	key := []byte(secret)
+	h := hmac.New(sha256.New, key)
+	h.Write([]byte(message))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func main() {
+	tmpl := template.Must(template.ParseFiles("layout.html"))
```

Comment:
Are these tabs? Does `go format` use *tabs*?

Reply:
What is `go format`?

Reply:
The go docs say `We use tabs for indentation and gofmt emits them by default. Use spaces only if you must.`